botocore/endpoint.py

```diff
@@ -152,16 +130,13 @@ def _create_request_object(self, request_dict):
                              headers=headers)
         return request
 
-    def prepare_request(self, request, signer):
-        if signer is not None:
-            with self._lock:
-                # Parts of the auth signing code aren't thread safe (things
-                # that manipulate .auth_path), so we're using a lock here to
-                # prevent race conditions.
-                event_name = 'before-auth.%s' % self._endpoint_prefix
-                self._event_emitter.emit(
-                    event_name, endpoint=self, request=request, auth=signer)
-                signer.add_auth(request=request)
+    def prepare_request(self, request, operation_model=None):
+        if operation_model:
+            event_name = 'request-created.{endpoint_prefix}.{op_name}'.format(
```

Comment:
Spoke in person, and I'm going to refactor this to create a request object for each retry. Then the event will be moved into `create_request` and everything will make sense, plus we have less of a chance of stale values causing issues later on.